packages/tupaia-web-server/src/routes/DashboardsRoute.ts

```diff
@@ -21,22 +26,62 @@ export class DashboardsRoute extends Route<DashboardsRequest> {
       })
     )[0];
 
-    const baseEntity = await ctx.services.entity.getEntity(
+    // We're including the root entity in this request, so we don't need to double up fetching it
+    const entities = await ctx.services.entity.getAncestorsOfEntity(
       project['entity_hierarchy.name'],
       entityCode,
+      {},
+      true,
     );
+    const rootEntity = entities.find((e: Entity) => e.code === entityCode);
 
     const dashboards = await ctx.services.central.fetchResources('dashboards', {
-      filter: { root_entity_code: baseEntity.code },
+      filter: { root_entity_code: entities.map((e: Entity) => e.code) },
```

Comment:
Appreciate you getting all this going using the central-api. If it ever feels too complex or that we're not keeping things DRY enough, I think we can consider giving `tupaia-web-server` access to some of the models here (eg. dashboard, dashboardItem, etc.). Being able to use of some the utility functions like `getDashboards` could be useful.

The big pro of just using the central-api is that we keep permissions centralised which is really nice. So let's see how things go

Reply:
Yeah I've considered it but I also feel like that sorta betrays the core tenant of an orchestration server. I think generally it's nicer to extend functionality of our micro-services (which is why in a case like this I extended the entity server) and keep the orchestration servers using minimal DB fetches